Noah - f2pHgty8hw/Week10_Assignment.do

```diff
@@ -0,0 +1,247 @@
+///////////////////////////////////////
+///*** ECON 490 WEEK 10 HOMEWORK ***///
+///////////////////////////////////////
+
+*Name: Noah Blake Smith
+
+*Date: November 13, 2022
+
+clear all
+set seed 1
+set more off
+
+global user "/Users/nbs/Desktop/week_10/03_assignment/01_data/" // User should to appropriate folderpath
+
+////////////////////////
+///*** QUESTION 1 ***///
+////////////////////////
+
+*Set globals
+global civ_density "$user/CIV_populationdensity.xlsx"
+global civ_section0 "$user/CIV_Section_0.dta"
+
+*Import data
+import excel "$civ_density", sheet("Population density") firstrow clear
+
+*Include only department data
+keep if regex(NOMCIRCONSCRIPTION,"DEPARTEMENT")
+
+*Extract relevant information and clean up
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION,"DEPARTEMENT DE","",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION,"DEPARTEMENT D'","",.)
+replace NOMCIRCONSCRIPTION = subinstr(NOMCIRCONSCRIPTION,"DEPARTEMENT DU","",.)
+replace NOMCIRCONSCRIPTION = lower(strtrim(stritrim(NOMCIRCONSCRIPTION)))
+
+*Rename variables
+ren NOMCIRCONSCRIPTION department
+ren DENSITEAUKM density
+ren *, lower
+
+*Fix typo
+replace department="arrha" if department=="arrah"
+
+*Create and save tempfile
+tempfile density
+save `density'
+
+*Import, clean, and merge CIV data
+use "$civ_section0", clear
+decode b06_departemen, generate(department)
+merge m:1 department using `density'
+
+*Check for errors
+tabulate department if _merge == 1 // No observations in civ_section0 are unmatched_
+tabulate department if _merge == 2 // One district not in civ_section0 called "gbeleban" (problematic but not world-ending)
+
+*Outro
+compress
+save "question1.dta", replace
+
+////////////////////////
+///*** QUESTION 2 ***///
+////////////////////////
+
+*Setup
+clear all
+global gps_data "$user/GPS data.dta"
+use "$gps_data", clear
+
+*ssc install geodist // User-created command to calculate distances
+
+gen j = 100000 // Arbitrary number for loop below
+local i = 0 // Counter for loop below
+
+while j[1] > 74100 { // Arbitrary threshold I chose that seemed reasonable
+	
+	local i = `i' + 1 // Update then display loop number
+	di `i'
+	
+	cap drop longitude0 latitude0 distance longitude_* latitude_* group length height area criterion // Drop variables from previous loop
+	
+	*Choose random coordinate
```

Comment:
That's a pretty clever way to do things, I just picked points and took the 5 other points closest to them.